src/internal/HMDataCheckList.cpp

```diff
@@ -20,7 +20,7 @@
 using namespace std;
 
 HM_SCHEDULE_STATE
-HMDataCheckList::checkNeeded(string& hostname, HMIPAddress& ip, HMDataHostCheck& hostCheck)
+HMDataCheckList::checkNeeded(string& hostname, HMIPAddress& ip, HMDataHostCheck& hostCheck, uint32_t version)
```

Comment:
can you add the default parameter like in header uint32_t version = HM_DEFAULT_STATE_VERSION / 0